app/models/setting.rb

```diff
@@ -83,4 +84,38 @@ def self.invoice_reminder_in_days
   def self.wishlist_size
     Setting.find_by(code: :wishlist_size).value.to_i
   end
+
+  def self.wishlist_supported_domain_extensions
+    extensions = Setting.find_by(code: :wishlist_supported_domain_extensions)
+    if extensions.present?
+      return JSON.parse(extensions.value) if json_array?(extensions.value)
+    end
+    []
+  end
+end
+
+private
+
+def valid_format_for_wishlist_domain_extensions
+  return unless code == 'wishlist_supported_domain_extensions'
+  return if valid_domain_extension_elements?(value)
+
+  errors.add(:value, 'is invalid')
+end
+
+def json_array?(text)
+  JSON.parse(text).is_a?(Array)
+rescue JSON::ParserError
+  false
+rescue NoMethodError
+  false
+end
+
+def valid_domain_extension_elements?(hash)
+  return false unless json_array?(hash)
+
+  r = /\A([a-z0-9\-\u00E4\u00F5\u00F6\u00FC\u0161\u017E]{1,61}\.)?[a-z0-9]{1,61}\z/x
```

Comment:
Not sure I understand this.

Reply:
Regex to check that domain extension is in valid format, without any whitespaces. Accepts xxx / xxx.xx

Reply:
Can we document this knowledge in the variable name? Like `domain_name_regexp`? If not, it's worth commenting it describing what does it match. Maybe like here https://github.com/internetee/auction_center/blob/df6009fbc588cf7605094e35b845100434c3ddab/app/models/wishlist_item.rb#L13 or in some other suitable way.